nerfstudio/scripts/datasets/process_project_aria.py

```diff
@@ -0,0 +1,231 @@
+    assert nearest_pose_idx != -1, f"Could not find pose for {capture_time_ns}"
+    t_world_device = t_world_devices.t_world_devices[nearest_pose_idx]
+
+    # Compute the world to camera transform.
+    t_world_camera = t_world_device @ camera_calibration.t_device_camera @ T_ARIA_NERFSTUDIO
+
+    return AriaImageFrame(
+        camera=camera_calibration,
+        file_path=file_path,
+        t_world_camera=t_world_camera,
+        timestamp_ns=capture_time_ns,
+    )
+
+
+def to_nerfstudio_frame(frame: AriaImageFrame) -> Dict:
+    return {
+        "fl_x": frame.camera.fx,
+        "fl_y": frame.camera.fy,
+        "cx": frame.camera.cx,
+        "cy": frame.camera.cy,
+        "distortion_params": frame.camera.distortion_params.tolist(),
+        "w": frame.camera.width,
+        "h": frame.camera.height,
+        "file_path": frame.file_path,
+        "transform_matrix": frame.t_world_camera.to_matrix().tolist(),
+        "timestamp": frame.timestamp_ns,
+    }
+
+
+@dataclass
+class ProcessProjectAria:
+    """Processes Project Aria data i.e. a VRS of the raw recording streams and the MPS attachments
+    that provide poses, calibration, and 3d points. More information on MPS data can be found at:
+      https://facebookresearch.github.io/projectaria_tools/docs/ARK/mps.
+    """
+
+    vrs_file: Path
+    """Path to the VRS file."""
+    mps_data_dir: Path
+    """Path to Project Aria Machine Perception Services (MPS) attachments."""
+    output_dir: Path
+    """Path to the output directory."""
+
+    def main(self) -> None:
+        """Generate a nerfstudio dataset from ProjectAria data (VRS) and MPS attachments."""
+        # Create output directory if it doesn't exist.
+        self.output_dir = self.output_dir.absolute()
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+
+        provider = create_vrs_data_provider(str(self.vrs_file.absolute()))
+        assert provider is not None, "Cannot open file"
+
+        name_to_camera = get_camera_calibs(provider)
+
+        print("Getting poses from closed loop trajectory CSV...")
+        trajectory_csv = self.mps_data_dir / "closed_loop_trajectory.csv"
+        t_world_devices = read_trajectory_csv_to_dict(str(trajectory_csv.absolute()))
+
+        name = "camera-rgb"
+        stream_id = provider.get_stream_id_from_label(name)
+
+        # create an AriaImageFrame for each image in the VRS.
+        print("Creating Aria frames...")
+        aria_frames = [
+            to_aria_image_frame(provider, index, name_to_camera, t_world_devices, self.output_dir)
+            for index in range(0, provider.get_num_data(stream_id))
+        ]
+
+        # create the NerfStudio frames from the AriaImageFrames.
+        print("Creating NerfStudio frames...")
+        CANONICAL_RGB_VALID_RADIUS = 707.5
+        CANONICAL_RGB_WIDTH = 1408
+        rgb_valid_radius = CANONICAL_RGB_VALID_RADIUS * (aria_frames[0].camera.width / CANONICAL_RGB_WIDTH)
```

Comment:
How did you get these numbers? On the project aria github linked [here](https://github.com/facebookresearch/projectaria_tools/blob/4aee633cb667ab927825dc10477cad0df8393a34/core/calibration/loader/SensorCalibrationJson.cpp#L97C1-L99C19), I see different numbers; Is there a reason why?

Reply:
Do you mean the width? This is just from different recording profiles; the RGB camera can be configured to record at either 1408x1408 (2MP) or 2880x2880 (8MP).

Reply:
Gotcha! that makes sense why the comment says "(full res during calibration)"